sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -48,29 +48,41 @@ import org.apache.spark.sql.util.SchemaUtils
  * @param properties the properties of this view.
  * @param originalText the original SQL text of this view, can be None if this view is created via
  *                     Dataset API.
- * @param child the logical plan that represents the view; this is used to generate the logical
- *              plan for temporary view and the view schema.
+ * @param plan the logical plan that represents the view; this is used to generate the logical
+ *             plan for temporary view and the view schema.
  * @param allowExisting if true, and if the view already exists, noop; if false, and if the view
  *                already exists, throws analysis exception.
  * @param replace if true, and if the view already exists, updates it; if false, and if the view
  *                already exists, throws analysis exception.
  * @param viewType the expected view type to be created with this command.
+ * @param isAnalyzed whether this command is analyzed or not.
  */
 case class CreateViewCommand(
     name: TableIdentifier,
     userSpecifiedColumns: Seq[(String, Option[String])],
     comment: Option[String],
     properties: Map[String, String],
     originalText: Option[String],
-    child: LogicalPlan,
+    plan: LogicalPlan,
     allowExisting: Boolean,
     replace: Boolean,
-    viewType: ViewType)
-  extends LeafRunnableCommand {
+    viewType: ViewType,
+    isAnalyzed: Boolean = false) extends RunnableCommand with AnalysisOnlyCommand {
 
   import ViewHelper._
 
-  override def innerChildren: Seq[QueryPlan[_]] = Seq(child)
+  override protected def withNewChildrenInternal(
+      newChildren: IndexedSeq[LogicalPlan]): CreateViewCommand = {
+    assert(!isAnalyzed)
+    copy(plan = newChildren.head)
+  }
+
+  override def innerChildren: Seq[QueryPlan[_]] = Seq(plan)
```

Comment:
shall we only include it as inner children only when `isAnalyzed = true`? Maybe we can put it in the base class, as it's better to include the plan in inner children to show it in the UI.

Reply:
@cloud-fan which UI are you referring to? If you are referring to the Spark UI, I see the following and it shows the same even if I use `override def innerChildren: Seq[QueryPlan[_]] = if (isAnalyzed) Seq(plan) else Nil`:
<img width="719" alt="Screen Shot 2021-04-23 at 9 30 57 PM" src="https://user-images.githubusercontent.com/12103644/115947249-42962180-a47b-11eb-90e6-874b592f97e1.png">

Reply:
how about the "optimized plan" in the EXPLAIN result?

Reply:
I see the following for the both cases. Were you expecting something different?

```
== Parsed Logical Plan ==
'CreateViewStatement [test], SELECT 1, false, false, PersistedView
+- 'Project [unresolvedalias(1, None)]
   +- OneRowRelation

== Analyzed Logical Plan ==

CreateViewCommand `default`.`test`, SELECT 1, false, false, PersistedView, true
   +- Project [1 AS 1#7]
      +- OneRowRelation

== Optimized Logical Plan ==
CreateViewCommand `default`.`test`, SELECT 1, false, false, PersistedView, true
   +- Project [1 AS 1#7]
      +- OneRowRelation

== Physical Plan ==
Execute CreateViewCommand
   +- CreateViewCommand `default`.`test`, SELECT 1, false, false, PersistedView, true
         +- Project [1 AS 1#7]
            +- OneRowRelation
```

Reply:
interesting. I thought there will be problems if a plan is in both `children` and `innerChildren`, but seems we are fine.

Reply:
Ah, the reason is that when explain runs, `CreateViewCommand` already went thru the `Analyzer` and its children is removed (note the `true` above, which is `isAnalyzed`).

Reply:
but the "Parsed Logical Plan" should be unresolved and the `plan` is in both `children` and `innerChildren`.

Reply:
"Parsed Logical Plan" is `CreateViewStatement`, not `CreateViewCommand`.

Reply:
ah now I get it. If there is a place that creates `CreateViewCommand` directly, we get a problem.

I think it's safer to avoid that to be future-proof. e.g.
```
override def innerChildren: Seq[QueryPlan[_]] = if (analyzed) Seq(plan) else Nil
```

Reply:
I agree. I will do a follow up PR.